daemon/cluster/swarm.go

```diff
@@ -430,6 +430,14 @@ func (c *Cluster) Info() types.Info {
 		info.NodeID = state.swarmNode.NodeID()
 	}
 
+	clusterInfo.WithValues(
+		info.Cluster.ID,
+		info.NodeID,
+		info.NodeAddr,
+	).Set(1)
+	clusterManagers.Set(float64(info.Managers))
+	clusterNodes.Set(float64(info.Nodes))
```

Comment:
@rogaha - could create a custom collector that calls `state.controlClient.ListNodes` as needed when the metrics are scraped:
https://godoc.org/github.com/prometheus/client_golang/prometheus#Collector

see especially the example there.

prometheus has a set of "*Func" metrics that allow you to specify a function to be called at metric collect time (ie, when the metrics are being scraped).